src/exml_query.erl

```diff
@@ -118,17 +127,20 @@ child_with_ns([#xmlel{} = Element | Rest], NS, Default) ->
 child_with_ns([_ | Rest], NS, Default) ->
     child_with_ns(Rest, NS, Default).
 
--spec subelement_with_attr(exml:element(), AttrName :: binary(), AttrValue :: binary()) ->
-    exml:element() | undefined.
+-spec subelement_with_attr(exml:element(),
+                           AttrName :: binary(),
+                           AttrValue :: binary()) -> exml:element() | undefined.
 subelement_with_attr(Element, AttrName, AttrValue) ->
     subelement_with_attr(Element, AttrName, AttrValue, undefined).
 
--spec subelement_with_attr(Element, AttrName, AttrValue, Other) -> SubElement | Other when
-      Element :: exml:element(),
-      AttrName :: binary(),
-      AttrValue :: binary(),
-      SubElement :: exml:element(),
-      Other :: term().
+-spec subelement_with_attr(Element, AttrName, AttrValue, Other) -> SubElement |
+                                                                   Other when Element ::
```

Comment:
`when` handling seems to be completely broken.
Expected: `when` followed immediately by a line break should reset indentation to the depth of the function name:

```
-spec subelement_with_attr(Element, AttrName, AttrValue, Other) -> SubElement | Other when<NEWLINE>
      Element :: exml:element(),
```

`when` followed by text should reset the following lines indentation to one character after `when`:

```
-spec subelement_with_attr(Element, AttrName, AttrValue, Other) -> SubElement | Other
      when Element :: exml:element(),
           AttrName :: binary(),
```

Reply:
@erszcz The formatter doesn't know anything about your indentation or characters. It obtains the AST of the module and, based on it, produces a new _formatted_ version of the same thing.
It's not possible to add rules like _if there is an enter in my code… do this, if not… do that_.
That said, feel free to open a ticket on the formatter repo specifying how you want _all_ specs with `when`s to be formatted and we can work on that.

Reply:
In general, formatters might work based on different principles. For example, Emacs/Vim formatters usually work on the text level, not the AST level - that's why I took the "procedural" approach of working with text when describing this issue.

Thanks for the explanation, though :+1: